File: ms-notification/src/main/java/com/guilherme/delfino/pubsub/notification/domain/entity/Event.java
package com.guilherme.delfino.pubsub.notification.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Event {
    private String title;
    private String description;
    private LocalDateTime date;
    private String location;
    private String authorName;
}

File: pub-sub-next/src/core/domain/entities/Event.ts
type EventParams = {
  id: string;
  title: string;
  description: string;
  author: string;
  date: Date;
  location: string;
};
export class Event {
  private _params: EventParams;
  constructor(params: Required<EventParams>) {
    this._params = params;
  }
  public get id() {
    return this._params.id;
  }
  public get title() {
    return this._params.title;
  }
  public get description() {
    return this._params.description;
  }
  public get author() {
    return this._params.author;
  }
  public get date() {
    return this._params.date;
  }
  public get location() {
    return this._params.location;
  }
}
